datafusion/physical-expr/src/planner.rs

```diff
@@ -43,29 +43,17 @@ use std::sync::Arc;
 /// * `e` - The logical expression
 /// * `input_dfschema` - The DataFusion schema for the input, used to resolve `Column` references
 ///                      to qualified or unqualified fields by name.
-/// * `input_schema` - The Arrow schema for the input, used for determining expression data types
-///                    when performing type coercion.
 pub fn create_physical_expr(
     e: &Expr,
     input_dfschema: &DFSchema,
-    input_schema: &Schema,
```

Comment:
I wonder why it is added at the beginning. For schema itself, could it be different between `input_dfschema` and `input_schema`?

Reply:
If no, then it is okay.

Reply:
it should be okay, if difference happen by any reason, we usually hit the error on planning `Error during planning: Mismatch between schema and batches`

Like below
https://github.com/apache/arrow-datafusion/issues/5695#issuecomment-1483877234

Reply:
Then it is weird to have duplicate parameters. I guess it might has some reason (i.e., rare cases they might be different).

@liukun4515 Do you remember it?

Reply:
I was able found an issue a year ago https://github.com/apache/arrow-datafusion/issues/4766 but later (https://github.com/apache/arrow-datafusion/issues/5695#issuecomment-1483550835) for similar query we started to see `Error during planning: Mismatch between schema and batches`. I can assume the check went to planning side.

Another suggestion on why it was done on physical expr side is we probably tried to ensure that schema are the same after all the optimization rules.